File: src/Laundro/Laundro.Core/Features/UserContextState/Models/UserContext.cs
using Laundro.Core.Domain.Entities;

namespace Laundro.Core.Features.UserContextState.Models;
public class UserContext
{
    public int UserId { get; set; }
    public string? Email { get; set; }

    public Tenant? Tenant { get; set; }
    public Role? Role { get; set; }
    public bool IsTenantOwner { get; set; } = false;

    //[JsonInclude]
    public List<Store>? Stores { get; set; }

    public static string Key => nameof(UserContext);
}

File: src/client-app/src/models/userContext.ts
import Role from './role';
import Store from './store';
import Tenant from './tenant';

interface UserContext {
  userId: number;
  email: string;
  tenantName?: string;
  tenantGuid: string;
  isTenantOwner?: boolean;
  roleSystemKey?: string;
  stores?: Store[];
}

export default UserContext;
